api/src/test/java/org/openmrs/module/openconceptlab/client/OclClientTest.java

```diff
@@ -132,4 +133,37 @@ public void parseDateFromPath_shouldParseDate() throws Exception {
 		
 		assertThat(date, is(calendar.getTime()));
 	}
+
+	@Test
+	public void getUrlVersion_shouldGetUrlVersion() throws IOException {
+		String url = "https://api.openconceptlab.org/users/username/collections/collectionname/v1.0";
+		String subscriptionToken = "53fc72f0498a707a26e4d903c0f24c2db24d1e35";
+		String version = oclClient.getUrlVersion(url, subscriptionToken);
+		assertThat(version, is("v1.0"));
+	}
+
+	@Test
+	public void getExportUrl_shouldConstructCorrectExportUrlWhenVersionIsPassedToUrl() throws URIException {
+		String url = "https://api.openconceptlab.org/users/username/collections/collectionname/v1.0";
+		String collectionVersion = "v1.0";
+		GetMethod getMethod = oclClient.getExportUrl(url, collectionVersion);
+		assertThat(getMethod.getURI().toString(), is(url + "/export"));
+	}
+
+	@Test
+	public void getExportUrl_shouldConstructCorretExportUrlWhenVersionIsNotPassedToUrl() throws URIException {
+		String url = "https://api.openconceptlab.org/users/username/collections/collectionname/";
+		String collectionVersion = "v2.0";
+		GetMethod getMethod = oclClient.getExportUrl(url, collectionVersion);
+		assertThat(getMethod.getURI().toString(), is(url + "v2.0/export"));
+	}
+
+	@Test
+	public void removeLastUrlForwardSlashIfExist_shouldWorkAsExpected() {
+		String url1 = "https://api.openconceptlab.org/users/username/collections/collectionname/";

```

Comment:
Are the urls variables to be changed? Or constants?

Reply:
They are to be changed, the method removes the last '/' at the end of the url if it exists

Reply:
In this test method, i do not see any where you are changing the variables url1 and url2

Reply:
This method "removeLastUrlForwardSlashIfExist" is converting url1 to https://api.openconceptlab.org/users/username/collections/collectionname

and url2 remain the same since it doesn't end with '/'

Reply:
What is mean is that as far as the test method is concerned, they are constants. Correct?

Reply:
Yes I get it now converting them to constants..

Reply:
done!